parser/Parser/Parser.py

```diff
@@ -0,0 +1,174 @@
+        """Transform data in file to nodes in with neo4j"""
+        index_comm=0
+        index_stopwords=1
+        index_synonyms=1
+        language_code_prefix = re.compile('[a-zA-Z][a-zA-Z]:')
+
+        #header
+        header,next_line = self.header_harvest()
+        yield header
+
+        #the other entries
+        self.block_index = 1 #to count block
+        data = self.reinitialize_data()
+        for line in self.file_iter(next_line):
+
+            if line == '':
+                if data['id']:
+                    yield data
+                    data = self.reinitialize_data()
+                    self.block_index+=1
+            elif line[0]=="#":
+                index_comm+=1
+                data['comment'].append(self.normalizing(line,"en"))
+            elif 'stopword' in line:
+                data['id'] = "stopwords:"+str(index_stopwords)
+                index_stopwords+=1
+                data['value'] = self.add_line(line[10:])
+            elif 'synonym' in line:
+                data['id'] = "synonyms:"+str(index_synonyms)
+                index_synonyms+=1
+                data['value'] = self.add_line(line[9:])
+            elif line[0]=="<":
+                data['parent_tag'].append(self.add_line(line[1:]))
+            elif language_code_prefix.match(line):
+                if not data['id'] : data['id']=self.add_line(line.split(',')[0])
+                #add tags and tagsid
+                lang=line[:2]
+                tags_list=[]
+                for word in line[3:].split(','):
+                    word_normalized=self.normalizing(word,lang)
+                    tags_list.append(word_normalized)
+                    data['tagsid'].append(lang+':'+word_normalized)
+                data['tags'][lang]=tags_list
+            else:
+                property_name, property_value = self.normalizing(line[:line.index(":")],"en"),add_line(line[line.index(":")+1:])
+                property={"id":len(data['properties'])+1,"name":property_name,"value":property_value}
+                data['properties'].append(str(property))
+
+    def txt2nodes(self):
+        """Adding nodes to database"""
+        harvested_data = self.harvest()
+        self.create_headernode(next(harvested_data))
+        for entry in harvested_data:
+            self.create_node(entry)
+
+    def parent(self):
+        """Create the relations between nodes"""
+        query = "match(n) WHERE size(n.parents)>0 return n.id, n.parents"
+        results = self.session.run(query)
+        for result in results:
+            id=result["n.id"]
+            parent_list=result["n.parents"]
+            for parent in parent_list:
+                yield parent,id
+
+    def create_child_link(self):
+        for parent_id,child_id in self.parent_search():
+            query="MATCH(p) WHERE '" + parent_id + "' IN p.tagsid" + " match(c) WHERE c.id='" + child_id + "' CREATE (c)-[:is_child_of]->(p)"
+        self.session.run(query)
+
+    def __call__(self):
```

Comment:
Instead of having filename as an attribute to the class, I would propose you to have it as an argument to this method, and pass it to txt2nodes.